Flaky-test: PersistentStreamingDispatcherBlockConsumerTest. testBlockBrokerDispatching

PersistentStreamingDispatcherBlockConsumerTest is flaky. The testBlockBrokerDispatching test method fails sporadically.

[example failure](https://github.com/apache/pulsar/runs/4259542832?check_suite_focus=true)

```
  Error:  testBlockBrokerDispatching(org.apache.pulsar.broker.service.persistent.PersistentStreamingDispatcherBlockConsumerTest)  Time elapsed: 10.142 s  <<< FAILURE!
1148
  org.testng.internal.thread.ThreadTimeoutException: Method org.apache.pulsar.client.api.DispatcherBlockConsumerTest.testBlockBrokerDispatching() didn't finish within the time-out 10000
```

<details>
<summary>Full exception stacktrace</summary>
<code><pre>
Error:  testBlockBrokerDispatching(org.apache.pulsar.broker.service.persistent.PersistentStreamingDispatcherBlockConsumerTest)  Time elapsed: 10.142 s  <<< FAILURE!
1148
  org.testng.internal.thread.ThreadTimeoutException: Method org.apache.pulsar.client.api.DispatcherBlockConsumerTest.testBlockBrokerDispatching() didn't finish within the time-out 10000
1149
  	at org.testng.internal.MethodInvocationHelper.invokeWithTimeoutWithNewExecutor(MethodInvocationHelper.java:371)
1150
  	at org.testng.internal.MethodInvocationHelper.invokeWithTimeout(MethodInvocationHelper.java:282)
1151
  	at org.testng.internal.TestInvoker.invokeMethod(TestInvoker.java:605)
1152
  	at org.testng.internal.TestInvoker.retryFailed(TestInvoker.java:214)
1153
  	at org.testng.internal.MethodRunner.runInSequence(MethodRunner.java:58)
1154
  	at org.testng.internal.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:822)
1155
  	at org.testng.internal.TestInvoker.invokeTestMethods(TestInvoker.java:147)
1156
  	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:146)
1157
  	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:128)
1158
  	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
1159
  	at org.testng.TestRunner.privateRun(TestRunner.java:764)
1160
  	at org.testng.TestRunner.run(TestRunner.java:585)
1161
  	at org.testng.SuiteRunner.runTest(SuiteRunner.java:384)
1162
  	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:378)
1163
  	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:337)
1164
  	at org.testng.SuiteRunner.run(SuiteRunner.java:286)
1165
  	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:53)
1166
  	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:96)
1167
  	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1218)
1168
  	at org.testng.TestNG.runSuitesLocally(TestNG.java:1140)
1169
  	at org.testng.TestNG.runSuites(TestNG.java:1069)
1170
  	at org.testng.TestNG.run(TestNG.java:1037)
1171
  	at org.apache.maven.surefire.testng.TestNGExecutor.run(TestNGExecutor.java:135)
1172
  	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeSingleClass(TestNGDirectoryTestSuite.java:112)
1173
  	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.executeLazy(TestNGDirectoryTestSuite.java:123)
1174
  	at org.apache.maven.surefire.testng.TestNGDirectoryTestSuite.execute(TestNGDirectoryTestSuite.java:90)
1175
  	at org.apache.maven.surefire.testng.TestNGProvider.invoke(TestNGProvider.java:146)
1176
  	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
1177
  	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
1178
  	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
1179
  	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)</pre></code>
</details>

@Jason918 [I also came across this type of stack trace](https://github.com/apache/pulsar/runs/4298924236?check_suite_focus=true#step:9:992) where the timeout didn't occur:
```
  Error:  Tests run: 16, Failures: 4, Errors: 0, Skipped: 12, Time elapsed: 24.47 s <<< FAILURE! - in org.apache.pulsar.broker.service.persistent.PersistentStreamingDispatcherBlockConsumerTest
  Error:  testBlockBrokerDispatching(org.apache.pulsar.broker.service.persistent.PersistentStreamingDispatcherBlockConsumerTest)  Time elapsed: 7.532 s  <<< FAILURE!
  java.lang.AssertionError: expected [600] but found [250]
  	at org.testng.Assert.fail(Assert.java:99)
  	at org.testng.Assert.failNotEquals(Assert.java:1037)
  	at org.testng.Assert.assertEqualsImpl(Assert.java:140)
  	at org.testng.Assert.assertEquals(Assert.java:122)
  	at org.testng.Assert.assertEquals(Assert.java:907)
  	at org.testng.Assert.assertEquals(Assert.java:917)
  	at org.apache.pulsar.client.api.DispatcherBlockConsumerTest.testBlockBrokerDispatching(DispatcherBlockConsumerTest.java:820)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
  	at java.base/java.lang.Thread.run(Thread.java:829)
```

--- a/pulsar-broker/src/test/java/org/apache/pulsar/client/api/DispatcherBlockConsumerTest.java
+++ b/pulsar-broker/src/test/java/org/apache/pulsar/client/api/DispatcherBlockConsumerTest.java
@@ -676,7 +676,7 @@ public class DispatcherBlockConsumerTest extends ProducerConsumerBase {
      * </pre>
      *
      */
-    @Test(timeOut = 10000)
+    @Test(timeOut = 60000)
     public void testBlockBrokerDispatching() {
         log.info("-- Starting {} test --", methodName);
 

